src/components/button/Button.jsx

```diff
@@ -0,0 +1,61 @@
+/**
+ * A reusable button component with customizable variants and sizes
+ *
+ * Supports different visual styles through the variant prop and multiple size options.
+ * Uses CSS custom properties for theming consistency across the application.
+ *
+ * @param {Object} props - Component props
+ * @param {string} props.text - Text content displayed inside the button
+ * @param {Function} props.onClick - Click event handler function
+ * @param {('primary')} [props.variant='primary'] - Visual style variant of the button
+ * @param {('sm'|'md'|'lg')} [props.size='md'] - Size of the button (small, medium, or large)
+ * @param {boolean} [props.disabled=false] - Whether the button is disabled
+ * @returns {JSX.Element} A styled button element
+ *
+ * @example
+ * // Basic usage with default props
+ * <Button text="Click me" onClick={() => console.log('Clicked!')} />
+ *
+ * @example
+ * // Large button with explicit variant
+ * <Button
+ *   text="Submit"
+ *   onClick={handleSubmit}
+ *   variant="primary"
+ *   size="lg"
+ * />
+ */
+function Button({ text, onClick, variant = "primary", size = "md", disabled = false }) {
```

Comment:
Passing the text like this won't render it.
```suggestion
function Button({ children, onClick, variant = "primary", size = "md", disabled = false }) {
```
Use the children prop from react.
Then you can use it like this when using the component
```jsx
<Button>Click me</Button>
<Button><Icon /> Save</Button>
```